### PR TITLE
Auto-convert each channel into own layer

### PR DESCRIPTION
**Pull request by youri-k**
This PR allows the auto-conversion to select one channel at a time for conversion to create multiple layer. Thus, more channel formats are supported. The supported format of 3x `uint8` channels is still converted into a single layer. Manually, one can specify the `--channel_index` option to enable channel selection as well.

Additionally, this PR also introduces the basic view configuration support, so the first 3 layer of a converted dataset can have different colors, when opened in webknossos.

### CODE REVIEW

File: wkcuber/converter.py
Comment (philippotto) [inline wkcuber/converter.py:379]:
```suggestion
                    args.layer_name = f"{layer_name}_{i}"
```

(note the `_`) should read a bit nicer in the final output

File: wkcuber/converter.py
Comment (philippotto) [inline wkcuber/converter.py:386]: can we unpack the other params, too, so that `args` hasn't to be passed additionally? at least, I'd pass `channel_index` separately to avoid mutation (then, parallelization over the channels also becomes easier).

File: wkcuber/cubing.py
Comment (philippotto) [inline wkcuber/cubing.py:96]:
```suggestion
        help="Select a single channel to be cubed into a layer. This can be helpful if several channels should be converted to multiple data layers. Note that webKnossos only supports multiple channels if these are three uint8 channels. If this is not the case, use --channel_index to create a layer per channel.",
```

a bit verbose, but hopefully clearer. i'm open for adaptations

**Review by philippotto — COMMENTED** [review]
only had a high-level glance at it. leaving the rest at @rschwanhold :)

File: wkcuber/api/Properties/LayerProperties.py
Comment (rschwanhold) [inline wkcuber/api/Properties/LayerProperties.py:118]: Unfortunately you also have to do this also for `SegmentationLayerProperties` (I don't know if there is a smarter way with inheritance)

**Review by rschwanhold — CHANGES_REQUESTED** [review]
Looks good overall, but I have added a few comments.

It would be nice if you can add a few tests:
- for the DatasetAPI it should be sufficient if you modify the `datasource-properties.json` in `test_dataset.py::test_read_and_write_of_properties`
- For the image reading it would be nice to have a test case that calls `read_image_file` for each channel separately and then compares it to the result of `read_image_file` for all channels at once. I have noticed that there are no test cases for the `image_reader` yet. So you can just create a new file `test_image_reader.py` and add an example image to `testdata/` (if there isn't a good image already). The test case should simply iterate over all channels and call `assert np.array_equal(...)`

File: wkcuber/image_readers.py
Comment (youri-k) [reply wkcuber/image_readers.py:188]: Thanks for the suggestion

File: wkcuber/image_readers.py
Comment (youri-k) [reply wkcuber/image_readers.py:189]: Yes, this was a false assumption because I changed it in the `cubing.py` file, but forgot to do it here. After merging the RGB TIFF fix, I reworked and reworded the variables. Please have another look

File: wkcuber/image_readers.py
Comment (youri-k) [reply wkcuber/image_readers.py:85]: Currently, our DM3 and DM4 reader only read single channel, so this should be already checked. Since the only channel one can select is `0`, this check should guard against all cases:
`assert (
            channel_index < num_channels
        ), "Selected channel is not present in the input files"`

File: wkcuber/cubing.py
Comment (youri-k) [reply wkcuber/cubing.py:250]: I added your suggestion for future users, but currently it's not needed because the image stack converter never gets called when no source files are found.

**Comment by youri-k** [comment]
Thank you @rschwanhold for your feedback. I only managed to fix half of the stuff today, but I will ping you, once the rest is done.

File: wkcuber/image_readers.py
Comment (rschwanhold) [reply wkcuber/image_readers.py:85]: Perfect

File: wkcuber/api/Properties/DatasetProperties.py
Comment (youri-k) [reply wkcuber/api/Properties/DatasetProperties.py:25]: Yes, this is intentional because there are dataset-wide settings (e.g. `position`) and settings per layer (e.g. `color`)

File: wkcuber/api/Properties/DatasetProperties.py
Comment (youri-k) [reply wkcuber/api/Properties/DatasetProperties.py:148]: Thanks for noticing the missing export. Since these values are webKnossos specific and only `wkw` can be read by webKnossos, I think it's fine to not support these values for `TiffDatasets`

**Comment by youri-k** [comment]
This should be ready for review again :slightly_smiling_face:

File: tests/test_image_readers.py
Comment (rschwanhold) [inline tests/test_image_readers.py:30]: Nice

**Review by rschwanhold — APPROVED** [review]
Thanks for this PR. I have looked at the new commits and my comments are now resolved. 
@philippotto does the new interface of `cubing` meet your expectations?

File: wkcuber/converter.py
Comment (philippotto) [reply wkcuber/converter.py:390]: This comment is marked as resolved, but `args` is still passed as a whole? Were there any problems which blocked this change or did it slip through the cracks?

**Comment by philippotto** [comment]
Looks good to me :) I left only one remark about the passing of `args` as a whole. If it's easy to change, please do so. If not, merge away :)

File: wkcuber/converter.py
Comment (youri-k) [reply wkcuber/converter.py:390]: I unpacked the `args`, but had to additionally pass them as a whole because of `with get_executor_for_args(executor_args) as executor:`. Additionally, the compression relies on the dtype detection of the cubing beforehand.

**Comment by youri-k** [comment]
@philippotto I just pushed another version where we explicitly create an extra namespace only for the `executor` args. This should do the trick :slightly_smiling_face:

**Comment by philippotto** [comment]
> @philippotto I just pushed another version where we explicitly create an extra namespace only for the `executor` args. This should do the trick

Great! Small change, but looks way cleaner :rocket:

**Pull request merged**
